src/utils/QueryUtils.ts

```diff
@@ -143,9 +143,10 @@ export class QueryUtils {
     };
   }
 
-  static quoteAndEscapeIfNeeded(str: string): string {
+  static quoteAndEscapeIfNeeded(str: string, forceEscape = false): string {
     Assert.isString(str);
-    return QueryUtils.isAtomicString(str) || (QueryUtils.isRangeString(str) || QueryUtils.isRangeWithoutOuterBoundsString(str))
+    return !forceEscape &&
+      (QueryUtils.isAtomicString(str) || QueryUtils.isRangeString(str) || QueryUtils.isRangeWithoutOuterBoundsString(str))
       ? str
       : QueryUtils.quoteAndEscape(str);
   }
```

Comment:
Instead of adding the `forceEscape` parameter to `quoteAndEscapeIfNeeded`, we could call `quoteAndEscape` directly.

Reply:
Yeah, I did consider putting the branching one function level above, but I took the decision to put it there to limit the branching.
Escaping directly where the string was coming was not an option either (it was my 1st attempt)